internal/provider/incident_alert_sources_data_source.go

```diff
@@ -0,0 +1,230 @@
+	"github.com/incident-io/terraform-provider-incident/internal/provider/models"
+)
+
+var (
+	_ datasource.DataSource              = &IncidentAlertSourcesDataSource{}
+	_ datasource.DataSourceWithConfigure = &IncidentAlertSourcesDataSource{}
+)
+
+func NewIncidentAlertSourcesDataSource() datasource.DataSource {
+	return &IncidentAlertSourcesDataSource{}
+}
+
+type IncidentAlertSourcesDataSource struct {
+	client *client.ClientWithResponses
+}
+
+type IncidentAlertSourcesDataSourceModel struct {
+	ID           types.String                              `tfsdk:"id"`
+	Name         types.String                              `tfsdk:"name"`
+	SourceType   types.String                              `tfsdk:"source_type"`
+	AlertSources []IncidentAlertSourcesDataSourceItemModel `tfsdk:"alert_sources"`
+}
+
+type IncidentAlertSourcesDataSourceItemModel struct {
+	ID           types.String                        `tfsdk:"id"`
+	Name         types.String                        `tfsdk:"name"`
+	SourceType   types.String                        `tfsdk:"source_type"`
+	SecretToken  types.String                        `tfsdk:"secret_token"`
+	Template     *models.AlertTemplateModel          `tfsdk:"template"`
+	JiraOptions  *models.AlertSourceJiraOptionsModel `tfsdk:"jira_options"`
+	EmailAddress types.String                        `tfsdk:"email_address"`
+}
+
+func (d *IncidentAlertSourcesDataSource) Configure(ctx context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(*IncidentProviderData)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Data Source Configure Type",
+			fmt.Sprintf("Expected *IncidentProviderData, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+		return
+	}
+
+	d.client = client.Client
+}
+
+func (d *IncidentAlertSourcesDataSource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_alert_sources"
+}
+
+func (d *IncidentAlertSourcesDataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	var data IncidentAlertSourcesDataSourceModel
+	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Get all alert sources
+	result, err := d.client.AlertSourcesV2ListWithResponse(ctx)
+	if err == nil && result.StatusCode() >= 400 {
+		err = fmt.Errorf("%s", string(result.Body))
+	}
+	if err != nil {
+		resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to list alert sources, got error: %s", err))
+		return
+	}
+
+	// Filter alert sources based on provided criteria
+	var filteredSources []client.AlertSourceV2
+	for _, source := range result.JSON200.AlertSources {
+		// Apply filters if they are provided
+		if !data.ID.IsNull() && source.Id != data.ID.ValueString() {
```

Comment:
I think we can skip an ID filter, and add a distinct `alert_source` (singular) in the future instead